### DISCUSSION TITLE
Google Auth say app name

### DISCUSSION THREAD

**Discussion by lukejgaskell** [supabase/supabase#2532] [Questions] [answered]
Hey everyone, I am trying to use google auth and it displays to the user "superbase.io" is the app requesting the permissions. Is there a way to get it say my app instead?

**Comment by dshukertjr** [comment]
To clarify, you are referring to this `supabase.co`, correct?
<img height="400" src="https://user-images.githubusercontent.com/18113850/127069859-40a2e30d-bea3-40dd-9767-2201c997cdd9.png" alt="google signin" />

Unfortunately, that domain is the domain used to sign in, so it cannot be changed at the moment. I found a SO answer about Firebase with similar problem. They are able to point `CNAME` record of the custom domain to firebase auth domain and use that to sign in user. They are able to do that, because Firebase provides SSL to custom domain through Firebase hosting. As of right now, Supabase does not provide SSL for custom domains, so at the moment, changing the auth domain seems not possible. 

There is an issue aiming to solve this problem, so hopefully this becomes possible in the future! 
https://github.com/supabase/gotrue/issues/143

**Reply by lukejgaskell** [reply]
Yes, that's exactly the message I'm talking about. The `CNAME` solution sounds viable. Thanks for linking to that enhancement!

**Comment by jeremyrajan** [comment]
Yea, this is something I am a bit confused how to go about. Because it will mislead the user when using google auth and they may not continue. Is there any workaround for this or all supabase customers will have this issue?

Thanks!

**Comment by kachar** [comment]
In my scenario, it says the full supabase app subdomain 

![image](https://user-images.githubusercontent.com/893608/149671637-21db7262-5944-442e-b4b1-de7a3d26995b.png)

Waiting to see how this issue develops in order to update our app

Here's the collection of links that concern the topic:

- https://github.com/supabase/gotrue/issues/143
- https://github.com/supabase/gotrue/issues/142
- https://github.com/supabase/supabase/issues/1895

**Reply by miketromba** [reply]
Another related discussion:
- https://github.com/supabase/supabase/discussions/2925

**Comment by eliranbenishai** [comment]
In one of the threads it's mentioned that this issue was targeted for Q2-3 of this year. Q4 is nearing its end and no update was given on this anywhere. Anyone knows where it stands? 
I've disabled OAUTH logins for now because the scammy looking "... continue to xxxx.supabase.co" is not something that can be taken seriously. :(
Thanks.

**Reply by relferreira** [reply]
@eliranbenishai  This thread might help you
https://github.com/supabase/supabase/discussions/2925

In my case, I went through this verification process, and now it shows my app's name on the OAuth log-in page.

**Comment by ehanahamed** [comment, marked as answer]
There is a way to solve this, (see #2925)

## Step 1: Configuring stuff in Google Cloud Developer Console Thingy

Go to the `OAuth consent screen` section

![2023-06-21_16-31](https://github.com/supabase/supabase/assets/98765025/f1955d3b-65cf-4e06-8cce-fd52abd18437)

Make sure you fill in the `App name`, `Application home page`, etc (example screenshots from one of my projects)

![2023-06-21_16-33](https://github.com/supabase/supabase/assets/98765025/01251202-4759-4a04-a278-46a53d22b662)

![2023-06-21_16-36](https://github.com/supabase/supabase/assets/98765025/ccc9acd6-cded-4bc2-baa0-85d0235c2f87)

Then, after all your app's information is filled, go to `Publishing status`, and set it from `testing` to `production`

![2023-06-21_16-38](https://github.com/supabase/supabase/assets/98765025/b6792917-553e-4849-b253-9fb92780963f)

It may ask you for some extra information, fill all of that in, and after your done go back to the `OAuth consent screen` section, it should now look like this:

![2023-06-21_16-39](https://github.com/supabase/supabase/assets/98765025/facc71f7-4a3c-47d2-9cf8-b058c4fbee89)

## Step 2: Waiting for an email from the google people

As the screen before said, you need to wait a long time for an email from google. I got my email about 2 weeks after i submitted the form thingy from the previous step.

When you get that email, they'll probably tell you that `"yOu DiDNt vEriFY oNE oF yOuR DOmAinS!!!"`

You probably DID verify all of ***your*** domains, but one of the domains in the `authorized domains` sections is `"xxxxxx.supabase.co"`. You cannot verify that domain, cause its not yours lol.

![2023-06-21_16-44](https://github.com/supabase/supabase/assets/98765025/710feae6-73b0-43b8-a45b-ed23f5fd288d)

Just email them back, telling them `"My app uses a third party redirect that I do not own"` (aka supabase), and they'll probably approve your app.

Theres a small chance they won't ask you about supbase (sometimes an actual human from the trust and saftey team will look at the url and realize its a third party callback), in that case they'll approve your app or will tell you to fix something else thats unrelated.

**Comment by Akash187** [comment]
I just want to update here that I got my verification done in 2-3 days. I didn't ask for approval for sensitive or restricted scopes.

**Comment by yiduoke** [comment]
https://supabase.com/docs/guides/platform/custom-domains

you have to pay $10/month to get the custom domain for your supabase. i did it and it worked
